PROTOCOL.md

```diff
@@ -515,11 +570,18 @@ To achieve the requirements above, related actions from different delta files ne
  - To replay the log, scan all file actions and keep only the newest reference for each logical file.
  - `add` actions in the result identify logical files currently present in the table (for queries). `remove` actions in the result identify tombstones of logical files no longer present in the table (for VACUUM).
 
+# Table Features
+Table Features must only exist on tables that have a supported protocol version. When the table's Reader Version is 3, `readerFeatures` must exist in the `protocol` action, and when the Writer Version is 7, `writerFeatures` must exist in the `protocol` action. `readerFeatures` and `writerFeatures` define the features that readers and writers must implement in order to read and write this table.
+
+Readers and writers must not ignore Table Features when it’s present:
+- to read a table, readers must implement and respect all features listed in `readerFeatures`that have `status = enabled`; readers which support features that have `status = blocked` must not enable those features, while readers which do not support blocked features may still read the table;
+- to write a table, writers must implement and respect all features listed in `writerFeatures` that have `status = enabled`; writers which support features that have `status = blocked` must not enable those features, while writers which do not support blocked features may still write to the table.
```

Comment:
I think we should make it an explicit point that effectively all writers have to read the table log, so must satisfy readerfeatures as well.